libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3696,6 +3706,9 @@ string YulUtilFunctions::arrayConversionFunction(ArrayType const& _from, ArrayTy
 
 string YulUtilFunctions::cleanupFunction(Type const& _type)
 {
+	if (auto userDefinedValueType = dynamic_cast<UserDefinedValueType const*>(&_type))
+		return cleanupFunction(userDefinedValueType->underlyingType());
```

Comment:
It'd probably be more consistent to actually generate a function that's a thin wrapper around the underlying cleanup function similarly to the other proxy cases below.